library/src/data/index.js

```diff
@@ -41,20 +41,236 @@ window.CoNDeT.data = {
     },
 
     saveToFile: function (data) {
-      var element = document.createElement('a');
+      var element = document.createElement("a");
       element.setAttribute(
-        'href',
-        'data:text/plain;charset=utf-8,' +
+        "href",
+        "data:text/plain;charset=utf-8," +
           encodeURIComponent(JSON.stringify(data))
       );
-      element.setAttribute('download', 'CoNDeT.json');
+      element.setAttribute("download", "CoNDeT.json");
 
-      element.style.display = 'none';
+      element.style.display = "none";
       document.body.appendChild(element); //Required on firefox
 
       element.click();
 
       element.remove();
     },
   },
+
+  StateModifier: (function () {
+    constructor = function (state) {
+      this.stateManager = state;
+      var self = this;
+      state.subscribe(function (newState) {
+        self.state = newState;
+      });
+    };
+
+    moveTable = function (tableId, newCoordinates) {
+      var newState = copyTables(this.state);
+      this.state[tableId].coordinates = {
+        x: newCoordinates.x,
+        y: newCoordinates.y,
+      };
+
+      this.stateManager.setState(newState);
+    };
+
+    createTable = function (newData) {
+      // newData = {name, className, coordinates, columns, rows}
+      var newState = copyTables(this.state);
+      newState.push(newData);
+      this.stateManager.setState(newState);
+    };
+
+    editContent = function (tableId, cell, header, name, className) {
```

Comment:
Po przemyśleniach myślę, że:
1. Może jednak rozbić na oddzielne dla zmiany nazwy, klasy, komórki, warunku i decyzji
2. Jak user będzie zmieniał, to będzie zmieniał jeden warunek/decyzję/komórkę. Zmiana komórki działa w sumie dobrze,  tylko przekazywałbym { type: "conditions" | "decisions", id: "nazwa decyzji/warunku", value: "nowa wartość"  }
W przypadku `conditions` i `decisions` na wejście podawałbym analogicznie obiekt z type, id i value (albo rozbił na dwie funkcje: `editCondition` / `editDecision`

Sorry, że dopiero teraz piszę o tak dużych zmianach, ale dopiero teraz dokładniej to przemyślałem. Powinienem rozpisać co jaka funkcja bierze przed zleceniem taska